src/chunkserver/chunkserver-common/chunk_trash_manager_unittest.cc

```diff
@@ -0,0 +1,79 @@
+// chunk_trash_manager_unittest.cc
+// A fix for https://stackoverflow.com/q/77034039/10788155
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
+#pragma GCC diagnostic ignored "-Wstringop-overflow"
+#include <filesystem>
+#include <gtest/gtest.h>
+#pragma GCC diagnostic pop
+
+#include <fstream>
+
+#include "chunk_trash_manager.h"
+#include "errors/saunafs_error_codes.h"
+
+class ChunkTrashManagerTest : public ::testing::Test {
+protected:
+	static std::filesystem::path testDir;
+
+	void SetUp() override {
+		testDir = std::filesystem::temp_directory_path() / "chunk_trash_manager_test";
+		std::filesystem::create_directories(testDir);
+	}
+
+	void TearDown() override {
+		std::filesystem::remove_all(testDir);
+	}
+};
+
+std::filesystem::path ChunkTrashManagerTest::testDir;
+
+// Test getDeletionTimeString() to ensure it produces a UTC timestamp
+TEST_F(ChunkTrashManagerTest, GetDeletionTimeStringTest) {
+	ChunkTrashManager manager;
+	std::string timestamp = manager.getDeletionTimeString();
+	EXPECT_EQ(timestamp.size(), 14);  // Check that the timestamp length is correct (YYYYMMDDHHMMSS)
+	EXPECT_TRUE(std::all_of(timestamp.begin(), timestamp.end(), ::isdigit));  // Check that all characters are digits
+}
+
+TEST_F(ChunkTrashManagerTest, MoveToTrashValidFile) {
+	// Test the moveToTrash() method directly using ChunkTrashManager
+	ChunkTrashManager manager;
+	std::filesystem::path filePath = testDir / "test_file.txt";
+	std::ofstream(filePath) << "dummy content";
+
+	std::string deletionTime = manager.getDeletionTimeString();
```

Comment:
Consider not to use an instance to access static members:
ChunkTrashManager::getDeletionTimeString();